src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -358,9 +366,18 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
                 })
             } else {
                 None
+                /*
```

Comment:
I left this code commented out to highlight a point.
I believe that when building a path dependency, this function returns None in order to not "pollute the suffix" of the executable or the library.

However, this means that when building across features, it appears to build to the same executable/library name.

It's correctly separating the .fingerprint, but not the actual built executable.